packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -345,9 +345,11 @@ abstract class FlameChartState<T extends FlameChart,
             nodes: nodes,
             width: math.max(constraints.maxWidth, widthWithZoom),
             startInset: widget.startInset,
-            selectionNotifier: widget.selectionNotifier,
-            searchMatchesNotifier: widget.searchMatchesNotifier,
-            activeSearchMatchNotifier: widget.activeSearchMatchNotifier,
+            selectionNotifier: widget.selectionNotifier as ValueListenable<V>,
+            searchMatchesNotifier:
+                widget.searchMatchesNotifier as ValueListenable<List<V>>,
+            activeSearchMatchNotifier:
+                widget.activeSearchMatchNotifier as ValueListenable<V>,
```

Comment:
why are these type casts necessary? searchMatchesNotifier and activeSearchMatchNotifier should likely stay nullable

Reply:
These casts becomes unnecessary if I replace `extends State<T>` with `extends State<FlameChart<T, V>>' in the class header. 
However this result in number of upstream errors, that are also fixable, but result on set of other errors. How about leaving this this tech debt?

Reply:
I see. Okay let's leave the type casts but match the nullable types that the widget has for searchMatchesNotifier and activeSearchMatchNotifier

Reply:
done